jar-persistence/src/main/java/com/sixsq/slipstream/persistence/VmRuntimeParameterMapping.java

```diff
@@ -129,14 +143,25 @@ public static void insertVmInstanceMapping(RuntimeParameter instanceIdParameter)
 		RuntimeParameter vmstate = getRuntimeParameter(runUuid, nodeInstanceName, RuntimeParameter.STATE_VM_KEY);
 		RuntimeParameter hostname = getRuntimeParameter(runUuid, nodeInstanceName, RuntimeParameter.HOSTNAME_KEY);
 
-		VmRuntimeParameterMapping m = new VmRuntimeParameterMapping(instanceId, cloud, runOwner, name, nodeName,
-				nodeInstanceid, vmstate, hostname);
-		m.store();
+		List<VmRuntimeParameterMapping> mappings = getMappings(runUuid, name);
+		int mappings_size = mappings.size();
 
-		logger.info("Created VMRtpMap for instanceId=" + instanceId + ", owner=" + runOwner + ", name=" + name
-				+ " vmstate=" + vmstate + ", m.id=" + m.id);
-		logger.info("VmRuntimeParameterMapping loaded by id = " + VmRuntimeParameterMapping.find(cloud, instanceId));
+		if (mappings_size > 1) {
+			logger.warning("Found "+ mappings_size + " mapping(s) for runuuid '" + runUuid + "' and name '" + name
+					+ "'.");
+		}
+
+		if (mappings_size > 0) {
+			logger.info("The mapping for runuuid '" + runUuid + "' and name '" + name + "' already exist. " +
+					"Not inserting a new one.");
+		} else {
+			VmRuntimeParameterMapping m = new VmRuntimeParameterMapping(instanceId, cloud, runOwner, name, nodeName,
+					nodeInstanceid, vmstate, hostname);
+			m.store();
```

Comment:
Too minor remarks:
* mappings_size could be renamed in mappingsSize
* the message in second `if` could be changed to `Mapping(s) ... already exist. No need to insert a new one.` (in case 2 or more already exist).